src/Core/TileMesh.js

```diff
@@ -187,8 +187,8 @@ TileMesh.prototype.getCoordsForLayer = function getCoordsForLayer(layer) {
         } else {
             throw new Error('unsupported projection wms for this viewer');
         }
-    } else if (layer.protocol == 'tms') {
-        return OGCWebServiceHelper.computeTMSCoordinates(this, layer.extent);
+    } else if (layer.protocol == 'tms' || layer.protocol == 'xyz') {
+        return OGCWebServiceHelper.computeTMSCoordinates(this, layer.extent, layer.protocol == 'xyz');
```

Comment:
Even if it's equivalent, I'd prefer to add a origin (or yOrigin) property to these layer.
You could assign a default value in  `TMS_Provider.prototype.preprocessDataLayer` based on the protocol - but this would allow the user to explicity define it.

Reply:
So you would have here, if I understand correctly:
```js
} else if (layer.protocol == 'tms') {
    return OGCWebServiceHelper.computeTMSCoordinates(this, layer.extent, layer.origin);
```
Seems good to me as we drop two tests doing it like this.